AppController/djinn.rb

```diff
@@ -4315,31 +4154,80 @@ def update_hosts_info()
   end
 
 
-  # Writes new nginx configuration files for the App Engine applications
-  # hosted in this deployment. Callers should invoke this method whenever
-  # there is a change in the number of machines hosting App Engine apps.
-  def regenerate_nginx_config_files()
-    Djinn.log_debug("Regenerating nginx config files for App Engine apps")
+  # Writes new nginx and haproxy configuration files for the App Engine
+  # applications hosted in this deployment. Callers should invoke this
+  # method whenever there is a change in the number of machines hosting
+  # App Engine apps.
+  def regenerate_routing_config()
+    Djinn.log_debug("Regenerating nginx and haproxy config files for apps.")
     my_public = my_node.public_ip
     my_private = my_node.private_ip
     login_ip = get_login.private_ip
 
-    Djinn.log_debug("@app_info_map is #{@app_info_map.inspect}")
-    @apps_loaded.each { |app|
+    apps_to_check = @apps_loaded + RESERVED_APPS
+    apps_to_check.each { |app|
+      next if app == "none"
+
+      # Check that we have the application information needed to
+      # regenerate the routing configuration.
+      if @app_info_map[app].nil? or @app_info_map[app]['nginx'].nil? or
+          @app_info_map[app]['nginx_https'].nil? or
+          @app_info_map[app]['haproxy'].nil? or
+          @app_info_map[app]['appengine'].nil?
+        Djinn.log_debug("Skipping routing for #{app} since it is not yet running.")
```

Comment:
Do we only set the app_info_map for an app after an appserver is up? If not, then the "it is not yet running" part might be misleading. You could say something like:

"Skipping routing for #{app} until all parameters are set."